lib/utils.js

```diff
@@ -53,3 +53,31 @@ export async function loadGetInitialProps (Component, ctx) {
   }
   return props
 }
+
+export class LockManager {
```

Comment:
I'd like to use an existing lib. But the promise pool and others similar doesn't work. They are good, if we know the amount of jobs we've do.

In our case, we don't know how many prefetches we need and when they arrive. So, what we need is a continuously running job manager with a max cap for parallel  jobs.

Frankly, I couldn't find a something else.
Anyway, I like locks and would love to use them with JS.
Unlike other languages, memory accessing is thread safe :) So, I love it.

> Anyway, if you know a lib which does what we want, let's use it.